### PR TITLE
Added dev guidelines for azure actions

### PR DESCRIPTION
**Pull request by BALAGA-GAYATRI**
This is based on azure login action's dev cycle. The steps are written in a generic way so that it is applicable for the other azure actions as well. In those actions this pointer for guidance will be added next.

### CODE REVIEW

File: docs/developer-guildelines.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Contributing to the existing actions
```
Comment (kaverma) [inline docs/developer-guildelines.md:1]: We should add a link to this in the TOC in the readMe for this repo as well.

File: README.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ This repository provides a framework, guidleines and processes to author new and
   - [Starter Action Workflows to deploy to Azure](https://github.com/Azure/actions-workflow-samples)
 - Contributing to Azure Actions
   - [Create a new GitHub Action for Azure or Microsoft](docs/Process_of_Authoring_GitHub_Actions_for_Azure.md#creating-a-new-github-action-for-azure-or-microsoft)
+  - [Contributing to the existing existing actions](docs/developer-guildelines.md)
```
Comment (kaverma) [inline README.md:11]: typo - "existing  existing"

File: docs/developer-guildelines.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Run __“npm install”__ in the terminal within the repo. This command downloads dependencies defined in a __package. json__ file and generates a node_modules folder with the installed modules.
+
+1. Make the necessary changes to __.ts__ files in __src/__ which after compilation generate __.js__ files in __lib/__ .
+
+1. If there are significant changes in input and output fields in action, add/remove those fields in __action.yml__ file.
+
+1. Add unit tests and build. Run __"npm run test"__ to check if unit tests pass . Run __"npm run build"__ to build files and check in your files including node_modules to the forked repo.
+ 
+1. For end to end testing create a .yml workflow in __.github/workflows__ directory  and point it to your __org/repo@branch__ and test if your changes are working end to end. If your workflow depends on azure login, make sure to use azure login action in the workflow.If you are using the publish profile method, you may skip this step.
```
Comment (kaverma) [inline docs/developer-guildelines.md:13]: Please clarify this step more. Where to create a .yml workflow for testing - In a test repo?

File: docs/developer-guildelines.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ 
+1. Once code is tested and unit tests are added, please raise a PR against the actual repo’s master branch.
+
+1. Include .ts files, .js files, unit tests and documentation change if any on README, action.yml. Do not add node_modules to PR.
```
Comment (kaverma) [inline docs/developer-guildelines.md:17]: We should add a separate action on Releasing the changes also. After the PR is merged, taking the changes to release branch, testing on that branch and then updating the version tag.

File: docs/developer-guildelines.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+## Contributing to the existing actions
+
+1. Fork the repo and pull it to the local master branch.
+
+1. Run __“npm install”__ in the terminal within the repo. This command downloads dependencies defined in a __package. json__ file and generates a node_modules folder with the installed modules.
+
+1. Make the necessary changes to __.ts__ files in __src/__ which after compilation generate __.js__ files in __lib/__ .
+
+1. If there are significant changes in input and output fields in action, add/remove those fields in __action.yml__ file.
```
Comment (kaverma) [inline docs/developer-guildelines.md:9]: nit: we can remove 'significant'. 'If there are any changes in input and .."

File: README.md (hunk 2 above)
Comment (BALAGA-GAYATRI) [reply README.md]: fixed it. Thanks for pointing out!

**Review by kaverma — CHANGES_REQUESTED** [review]
@BALAGA-GAYATRI - Added few comments, can you please have a look.

File: docs/developer-guildelines.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+## Contributing to the existing actions
+
+1. Fork the repo and pull it to the local master branch.
```
Comment (ds-ms) [inline docs/developer-guildelines.md:3]: to the local `main` branch.

File: docs/developer-guildelines.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Run __“npm install”__ in the terminal within the repo. This command downloads dependencies defined in a __package. json__ file and generates a node_modules folder with the installed modules.
```
Comment (ds-ms) [inline docs/developer-guildelines.md:5]: The numbering seems to be wrong through out.

Comment (ds-ms) [inline docs/developer-guildelines.md:5]: Before running `npm install` mention the minimum required versions of node and npm to be present in the machine

File: docs/developer-guildelines.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Make the necessary changes to __.ts__ files in __src/__ which after compilation generate __.js__ files in __lib/__ .
+
+1. If there are significant changes in input and output fields in action, add/remove those fields in __action.yml__ file.
+
+1. Add unit tests and build. Run __"npm run test"__ to check if unit tests pass . Run __"npm run build"__ to build files and check in your files including node_modules to the forked repo.
```
Comment (ds-ms) [inline docs/developer-guildelines.md:11]: You're supposed to check in `node_modules` only in `releases/v*` branches

File: docs/developer-guildelines.md (hunk 3 above)
Comment (ds-ms) [reply docs/developer-guildelines.md:13]: Remove the section about publish profile, it's only specific to `web-apps`

File: docs/developer-guildelines.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+ 
+1. For end to end testing create a .yml workflow in __.github/workflows__ directory  and point it to your __org/repo@branch__ and test if your changes are working end to end. If your workflow depends on azure login, make sure to use azure login action in the workflow.If you are using the publish profile method, you may skip this step.
+ 
+1. Once code is tested and unit tests are added, please raise a PR against the actual repo’s master branch.
```
Comment (ds-ms) [inline docs/developer-guildelines.md:15]: `main` branch

File: docs/developer-guildelines.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    uses: username/reponame@branch
+    with:
+     input_A: 
+     input_B: 
+```
+where username/reponame@branch is your forked repo. Remember to checkin __node_modules__ in your forked repository while testing.
+
+8. Once code is tested and unit tests are added, please raise a PR against the actual repo’s main branch. Note that __node_modules__ should __not__ be pushed while raising PR against __main__ branch.
+
+9. Once your PR is merged, cherry pick your changes to __releases/v(latest)__ branch. After that PR is merged the repo maintainer will create a new release with a new tag. This release will be pre-release mode until certain bug bashes/regressive testing is done. Here for __releases/v(latest)__ branch, checkin your __node_modules__ if there is a new dependency in package.json.
```
Comment (kaverma) [inline docs/developer-guildelines.md:26]: Once your main branch PR is merged, cherry pick your changes to __releases/v(latest)__ branch and create a PR against the releases branch.

Comment (kaverma) [inline docs/developer-guildelines.md:26]: Please move this part to next step -
10. After that release PR is merged the repo maintainer will create a new release with a new tag. This release will be pre-release mode until certain bug bashes/regressive testing is done. Here for __releases/v(latest)__ branch, checkin your __node_modules__ if there is a new dependency in package.json.

**Review by kaverma — APPROVED** [review]

**Pull request merged**